lib/include/tick/array/array.h

```diff
@@ -526,48 +530,28 @@ CEREAL_LOAD_FUNCTION_NAME(Archive &ar, Array<T> &arr) {
  * @{
  */
 
-/**
- * @}
- */
-
-/** @defgroup array_sub_mod The instantiations of the Array template
- *  @ingroup Array_typedefs_mod
- * @{
- */
-
-/**
- * @}
- */
-
-/** @defgroup arraylist1d_sub_mod The classes for dealing with 1d-list of Arrays
- *  @ingroup Array_typedefs_mod
- * @{
- */
-
-/**
- * @}
- */
-
-/** @defgroup arraylist2d_sub_mod The classes for dealing with 2d-list of Arrays
- *  @ingroup Array_typedefs_mod
- * @{
- */
-
-#define ARRAY_DEFINE_TYPE(TYPE, NAME)                   \
+#define ARRAY_DEFINE_TYPE_BASIC(TYPE, NAME)             \
   typedef Array<TYPE> Array##NAME;                      \
   typedef std::vector<Array##NAME> Array##NAME##List1D; \
   typedef std::vector<Array##NAME##List1D> Array##NAME##List2D
 
+#define ARRAY_DEFINE_TYPE(TYPE, NAME)  \
+  ARRAY_DEFINE_TYPE_BASIC(TYPE, NAME); \
+  CEREAL_REGISTER_TYPE(Array##NAME)
+
 ARRAY_DEFINE_TYPE(double, Double);
```

Comment:
Can we just comment here that we use ARRAY_DEFINE_TYPE_BASIC for all types that don't need serialization yet?

Reply:
And add that's for compile time reasons

Reply:
sure,